pkg/remoteresolution/resolver/bundle/resolver.go

```diff
@@ -1,87 +1,144 @@
 type Resolver struct {
-	kubeClientSet kubernetes.Interface
+	kubeClientSet      kubernetes.Interface
+	resolveRequestFunc func(context.Context, kubernetes.Interface, *v1beta1.ResolutionRequestSpec) (resolutionframework.ResolvedResource, error)
 }
 
 // Initialize sets up any dependencies needed by the Resolver. None atm.
 func (r *Resolver) Initialize(ctx context.Context) error {
-	r.kubeClientSet = client.Get(ctx)
+	r.kubeClientSet = kubeclient.Get(ctx)
+	if r.resolveRequestFunc == nil {
+		r.resolveRequestFunc = bundleresolution.ResolveRequest
+	}
 	return nil
 }
 
 // GetName returns a string name to refer to this Resolver by.
-func (r *Resolver) GetName(context.Context) string {
+func (r *Resolver) GetName(_ context.Context) string {
 	return BundleResolverName
 }
 
-// GetConfigName returns the name of the bundle resolver's configmap.
-func (r *Resolver) GetConfigName(context.Context) string {
-	return bundle.ConfigMapName
-}
-
-// GetSelector returns a map of labels to match requests to this Resolver.
-func (r *Resolver) GetSelector(context.Context) map[string]string {
+// GetSelector returns a map of labels to match against tasks requesting
+// resolution from this Resolver.
+func (r *Resolver) GetSelector(_ context.Context) map[string]string {
 	return map[string]string{
-		common.LabelKeyResolverType: LabelValueBundleResolverType,
+		resolutioncommon.LabelKeyResolverType: LabelValueBundleResolverType,
 	}
 }
 
-// Validate ensures reqolution request spec from a request are as expected.
+// GetConfigName returns the name of the bundle resolver's configmap.
+func (r *Resolver) GetConfigName(_ context.Context) string {
+	return bundleresolution.ConfigMapName
+}
+
+// Validate ensures parameters from a request are as expected.
 func (r *Resolver) Validate(ctx context.Context, req *v1beta1.ResolutionRequestSpec) error {
-	if len(req.Params) > 0 {
-		return bundle.ValidateParams(ctx, req.Params)
+	return bundleresolution.ValidateParams(ctx, req.Params)
+}
+
+// IsImmutable implements ImmutabilityChecker.IsImmutable
+// Returns true if the bundle parameter contains a digest reference (@sha256:...)
+func (r *Resolver) IsImmutable(params []v1.Param) bool {
+	var bundleRef string
+	for _, param := range params {
+		if param.Name == bundleresolution.ParamBundle {
+			bundleRef = param.Value.StringVal
+			break
+		}
+	}
+
+	// Checks if the given string looks like an OCI pull spec by digest.
+	// A digest is typically in the format of @sha256:<hash> or :<tag>@sha256:<hash>
+	// Check for @sha256: pattern
+	if strings.Contains(bundleRef, "@sha256:") {
+		return true
 	}
-	// Remove this error once validate url has been implemented.
-	return errors.New("cannot validate request. the Validate method has not been implemented.")
+	// TODO(twoGiants): doesn't look right => write test => should check for two colons
+	// Check for :<tag>@sha256: pattern
+	if strings.Contains(bundleRef, ":") && strings.Contains(bundleRef, "@sha256:") {
+		return true
+	}
```

Comment:
I am not sure why this is here as it would have already been taken care of in the previous case ?